game/src/js/objects/ui/DOMModal.js

```diff
@@ -56,25 +57,34 @@ export default class DOMModal {
       });
     });
     // Process data into HTML
+    
     if(this.props.data) {
       const objResolve = function(rootObj, string) {
         let value = rootObj;
         let parts = string.split('.');
         while(parts.length != 0) {
           value = value[parts.shift()];
         }
+        
         return value;
       }
-      this.modal.querySelectorAll('*[data-value]').forEach(element => {
-        if(element.nodeName == 'INPUT') {
+      
+      this.modal.querySelectorAll('*[data-checked]').forEach(element => {
+        if(element.nodeName == 'INPUT' && element.type != 'checkbox') {
           element.value = objResolve(this.props.data, element.getAttribute('data-value'));
+          console.log('data-value' + element.getAttribute('data-value'));
+        }
+        else if (element.nodeName == 'INPUT' && element.type == 'checkbox') {
+          element.checked = this.props.data.swCheck;
```

Comment:
This should read the value provided in the `data-value` attribute via `objResolve`